lib/url.js

```diff
@@ -395,10 +395,10 @@ Url.prototype.parse = function parse(url, parseQueryString, slashesDenoteHost) {
     this.query = Object.create(null);
   }
 
-  var firstIdx = (questionIdx !== -1 &&
-                  (hashIdx === -1 || questionIdx < hashIdx) ?
-                  questionIdx :
-                  hashIdx);
+  var firstIdx =
+    questionIdx !== -1 && (hashIdx === -1 || questionIdx < hashIdx) ?
```

Comment:
nit: assign the condition to a variable, and fit it in one line (`useQuestionIdx`)